File: 01.JS-Basic/More Exercises/PB - More Exercises/Nested Conditional Statements/Flowers.js
function flowers(input) {

    let chrysanthemums = Number(input[0])
    let roses = Number(input[1])
    let tulips = Number(input[2])
    let season = input[3]
    let holiday = input[4]

    let total = 0.0

    switch (season) {
        case "Spring":
            if (holiday === "Y") {
                if (tulips > 7) {
                    total = ((chrysanthemums * 2 + roses * 4.1 + tulips * 2.5) * 1.15) * 0.95
                }
                else {
                    total = (chrysanthemums * 2 + roses * 4.1 + tulips * 2.5) * 1.15
                }
            }
            else {
                if (tulips > 7) {
                    total = (chrysanthemums * 2 + roses * 4.1 + tulips * 2.5) * 0.95
                }
                else { total = chrysanthemums * 2 + roses * 4.1 + tulips * 2.5 }
            }; break;

        case "Summer":
            switch (holiday) {
                case "Y":
                    total = (chrysanthemums * 2 + roses * 4.1 + tulips * 2.5) * 1.15; break;
                case "N":
                    total = chrysanthemums * 2 + roses * 4.1 + tulips * 2.5
            }; break;
        case "Autumn":
            switch (holiday) {
                case "Y":
                    total = (chrysanthemums * 3.75 + roses * 4.5 + tulips * 4.15) * 1.15; break;
                case "N":
                    total = chrysanthemums * 3.75 + roses * 4.5 + tulips * 4.15
            }; break;
        case "Winter":
            switch (holiday) {
                case "Y":
                    if (roses >= 10) {
                        total = ((chrysanthemums * 3.75 + roses * 4.5 + tulips * 4.15) * 1.15) * 0.9
                    }
                    else { total = (chrysanthemums * 3.75 + roses * 4.5 + tulips * 4.15) * 1.15 }; break;
                case "N":
                    if (roses >= 10) {
                        total = (chrysanthemums * 3.75 + roses * 4.5 + tulips * 4.15) * 0.9
                    }
                    else { total = chrysanthemums * 3.75 + roses * 4.5 + tulips * 4.15 }; break;
            }

    }
    if (chrysanthemums + roses + tulips > 20) {
        total = total * 0.8
    }

    console.log((total + 2).toFixed(2))

}
flowers(["3", "10", "9", "Winter", "N"])

File: JAVA-Basic/JavaBasic/src/ConditionalStatementsAdvanced/MoreExercise/Flowers.java
package ConditionalStatementsAdvanced.MoreExercise;

import java.util.Scanner;

public class Flowers {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int hrisantemes = Integer.parseInt(scan.nextLine());
        int roses = Integer.parseInt(scan.nextLine());
        int tulips = Integer.parseInt(scan.nextLine());
        String season = scan.nextLine();
        String holiday = scan.nextLine();
        double flowersPrice = 0.00;

        switch (season) {
            case "Spring":
                flowersPrice += (hrisantemes * 2 + roses * 4.1 + tulips * 2.5);
                if (holiday.equals("Y")) {
                    flowersPrice *= 1.15;
                }
                if (tulips > 7) {
                    flowersPrice *= 0.95;
                }
                break;
            case "Summer":
                flowersPrice += (hrisantemes * 2 + roses * 4.1 + tulips * 2.5);
                if (holiday.equals("Y")) {
                    flowersPrice *= 1.15;
                }
                break;
            case "Autumn":
                flowersPrice += (hrisantemes * 3.75 + roses * 4.5 + tulips * 4.15);
                if (holiday.equals("Y")) {
                    flowersPrice *= 1.15;
                }
                break;
            case "Winter":
                flowersPrice += (hrisantemes * 3.75 + roses * 4.5 + tulips * 4.15);
                if (holiday.equals("Y")) {
                    flowersPrice *= 1.15;
                }
                if (roses >= 10) {
                    flowersPrice *= 0.9;
                }
                break;
        }
        if (hrisantemes + tulips + roses > 20) {
            flowersPrice *= 0.8;
        }
        flowersPrice += 2;

        System.out.printf("%.2f", flowersPrice);
    }
}
